### ISSUE TITLE
Use Windows FSharp within Windows Subsystem for Linux

### ISSUE THREAD

**Issue by Tombert** [fsharp/vim-fsharp#97]
I was recently trying to run the vim-fsharp plugin inside WSL, and pointed my MSBuild executable and Fsharp interactive executable to the windows one (since WSL now allows you to call windows applications), and while it would start fsautocomplete, it didn't appear to do anything else. 

Would it be feasible to modify the plugin to work within WSL?

**Comment by kjnilsson** [comment]
Forgive my ignorance, what is WSL?

**Comment by Tombert** [comment]
@kjnilsson WSL = Windows Subsystem for Linux.  It is Microsoft's ability to run a mostly-full linux in Windows userspace; substantially nicer than Cygwin, since it will actually run Linux binaries without a recompilation.

**Comment by kjnilsson** [comment]
I don't have any such system to test on so support for WSL would have to come from a contribution. PR's welcome.

**Comment by enricosada** [comment]
@Tombert i use WSL. i'd like to test vim plugin there.
Are you sure you have `mono` installed? that's a requirement. the name is `fsautocomplete.exe`, but is a .net exe so need to run with mono

**Comment by Tombert** [comment]
@enricosada I do not have Mono installed; WSL can actually run native Windows apps now, so theoretically it should be able to run my native windows install instead of the simulated Mono version, right?

**Comment by enricosada** [comment]
yes, you can run `./fsautocomplete.exe --help` and that will run as .NET on windows.

v-- `fsautocomplete.exe` invoked from WLS

![image](https://user-images.githubusercontent.com/147243/32118337-80bb05f0-bb51-11e7-8bdf-533a858acc12.png)

problem is, FSAC doesnt check the OS atm for some behaviour it need to do differently.
just check if is run from `mono` or not.
so is not going to work like that, because need some really specific detection (app run in windows, but started in WSL... so should use unix behaviour). dunno how to detect that

as a workaround, try isntall mono on WSL. so vim run `mono fsautocmplete.exe` and things should work

**Comment by Tombert** [comment]
@enricosada how hard would it be to add a check for the host OS to see if it's WSL?  I'd be happy to do a PR if you point me in the right direction.

**Comment by enricosada** [comment]
@Tombert but when in WSL, running FSAC in windows, is a nice use case.
we can use all the windows .net tooling, and unix tools + unix vim.

can you open an issue in https://github.com/fsharp/FsAutoComplete/issues/ ? but we need a way to detect we are started from WLS..

**Comment by enricosada** [comment]
> check for the host OS to see if it's WSL

exactly. dunno :D.
fsac will be a windows process, running on windows (probably), but started from WSL linux.. so dunno

**Comment by kjnilsson** [comment]
Mostly this should be a matter of detecting how to run fsac somewhere near [here](https://github.com/fsharp/vim-fsharp/blob/master/ftplugin/fsharpvim.py#L76) and do so. Atm it first tries with `mono` then without.
